### DISCUSSION TITLE
Discord Operator?

### DISCUSSION THREAD

**Discussion by ghost** [apache/airflow#30154] [Q&A] [answered]
Is the Discord Operator no longer functioning? I've been trying to import it but keep getting a `ModuleNotFoundError: No module named 'airflow.providers.discord'`

**Comment by potiuk** [comment, marked as answer]
Maybe you do not have discord provider installed simply,

**Reply by ghost** [reply]
I'm also using the `PostgresToGCSOperator` and `GCSToBigQueryOperator` and I didn't have to install anything? I am running Airflow through `docker-compose` if that changes anything.

**Reply by ghost** [reply]
Oh, I think it's because those are in the core Airflow packages....

**Reply by potiuk** [reply]
No. because the google provider is one of the extras available by default https://airflow.apache.org/docs/docker-stack/index.html#docker-image-for-apache-airflow

The Apache Airflow image provided as convenience package is optimized for size, and it provides just a bare minimal set of the extras and dependencies installed and in most cases you want to either extend or customize the image. You can see all possible extras in [Reference for package extras](https://airflow.apache.org/docs/apache-airflow/stable/extra-packages-ref.html). The set of extras used in Airflow Production image are available in the [Dockerfile](https://github.com/apache/airflow/blob/2c6c7fdb2308de98e142618836bdf414df9768c8/Dockerfile#L37).

**Reply by ghost** [reply]
Oh, I see what you're saying. I see slack in `AIRFLOW_EXTRAS`. How do I go about installing Discord in this manner? I see the reference for package extras. Is this something I need to specify in the `docker-compose.yml` file or do I simply run `pip install 'apache-airflow[discord]'` in my terminal?

**Reply by potiuk** [reply]
read the docs I linked. you need to build new image.

**Reply by ghost** [reply]
Okay, thank you for your help.
